pkg/operator/controller/ingress/status.go

```diff
@@ -619,13 +619,27 @@ func computeLoadBalancerStatus(ic *operatorv1.IngressController, service *corev1
 		// Try and find a more specific reason for for the pending status.
 		createFailedReason := "SyncLoadBalancerFailed"
 		failedLoadBalancerEvents := getEventsByReason(operandEvents, "service-controller", createFailedReason)
-		for _, event := range failedLoadBalancerEvents {
-			involved := event.InvolvedObject
-			if involved.Kind == "Service" && involved.Namespace == service.Namespace && involved.Name == service.Name && involved.UID == service.UID {
-				reason = "SyncLoadBalancerFailed"
-				message = fmt.Sprintf("The %s component is reporting SyncLoadBalancerFailed events like: %s\n%s",
-					event.Source.Component, event.Message, "The kube-controller-manager logs may contain more details.")
-				break
+		if len(failedLoadBalancerEvents) > 0 {
+			for _, event := range failedLoadBalancerEvents {
+				involved := event.InvolvedObject
+				if involved.Kind == "Service" && involved.Namespace == service.Namespace && involved.Name == service.Name && involved.UID == service.UID {
```

Comment:
Just curious - did you happen to experiment to see if the filter on this was too strict, and we somehow lost messages that would have been relevant?

Reply:
There may be more checks than necessary, but I don't think it will reject any events that would be relevant. 

Specifically, maybe the namespace check is unnecessary since I think this is getting events in the `openshift-ingress` namespace, and all those events seem to have involved objects in the same namespace, but it's possible that's not a safe assumption on my part.